appinventor/components/src/com/google/appinventor/components/runtime/Sound.java

```diff
@@ -206,7 +211,7 @@ public void Source(@Asset String path) {
           }
         } catch (PermissionException e) {
           form.dispatchPermissionDeniedEvent(this, "Source", e);
-        } catch (IOException e) {
+        } catch (Exception e) {
```

Comment:
nit: Why change this `IOException` to `Exception`? I understand we may be interested in catching any other exception, but in this clause, we are throwing the "unable to load" error on App Inventor, so if any other exception happens now, they will receive "unable to load" despite it could be any other error.
Maybe we could keep `IOException` as it is, and just below we can catch a generic `Exception` and throw it in App Inventor as "unknown error" rather than "unable to load".

Reply:
I changed IOException to Exception because it had been changed in the Kodular-Creator codebase. See https://github.com/Kodular/Kodular-Creator/commit/9c5d6cd59dc4a414518704473b3ec471c858ab00
It's not clear from that commit why it was done. I can't trace that commit to a pull request.

There isn't an `unknown error` message yet and since the exception occurred when the it was trying to load, I think it's ok to use `unable to load`, but I don't feel strongly about it at all.

Here are the choices that I can think of:
1.  Change Exception back to IOException.
2.  Leave it as IOException with no further changes.
3.  Leave it as IOException and add a new error message for `unknown error`.

Reply:
That commit made `IOException` -> `Exception` changes in the code of 25 components.  Can anyone at Kodular explain why that extensive change was made?  Did something perhaps change with error handling at a higher level that necessitated that?  The answers to these might help Liz in choosing what to do.

Reply:
This change was made to catch all sorts of exceptions so as not to crash at runtime, if we ever missed any exception. It has no other significance.

Reply:
I think checked exceptions should be handled by the components, but any unchecked exceptions should probably be handled in the runtime instead. While it's not relevant in this particular case, some of the newer blocks throw `StopBlocksException`, which would be caught here and blocks would continue executing rather than bubbling up to runtime.scm. The other benefit of doing it in runtime.scm is that it helps centralize the error handling rather than having to make changes to many components.

Reply:
>While not a best practice, we had to do it to ship faster, as silently catching an exception is a better UX than crashing at runtime.

If there were exceptions in components that were causing the app to crash at runtime then I think that that probably indicated a bug somewhere in runtime.scm which, IIRC, is supposed to catch any (ideally unchecked) exceptions that get through and process them in `process-exception`

Reply:
One possibility is that the crashes were occurring in asynchronous routines, where an uncaught exception won't go through runtime.scm and will cause the app to crash. One way we could address that is by also adding some error handling logic to AsynchUtil, which I don't think has much, if any. In either case, I think that's a change independent of this PR.

Reply:
Thanks for the discussion. I think only handling checked exceptions in components is a good rule.
I've changed Exception back to IOException.